packages/paste-website/src/pages/blog/2025-03-20-css-variables-ssr-ssg.mdx

```diff
@@ -0,0 +1,189 @@
+
+Here is the script that is found in <Anchor showExternal={true} href="https://github.com/twilio-labs/paste/blob/main/packages/paste-website/src/pages/_document.tsx"><InlineCode>_documents.tsx</InlineCode></Anchor>:
+
+<Box marginBottom="space60">
+  <CodeBlockWrapper>
+  <CodeBlockHeader>Setting data-theme using cookie before render</CodeBlockHeader>
+    <CodeBlock language='javascript' code={`<script
+  type="text/javascript"
+  dangerouslySetInnerHTML={{
+    __html: \`
+      let parts = typeof document !== "undefined" && document?.cookie.split("paste-docs-theme=");
+      let cookie = parts.length == 2 ?parts?.pop().split(";").shift() : null;
+      if(cookie){
+        document.body.dataset.theme = cookie;
+      }
+      else if(window !== "undefined" && window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches){
+        document.body.dataset.theme = "twilio-dark";
+      }
+  \`,
+  }}
+/>`}/>
+  </CodeBlockWrapper>
+</Box>
+
+### _app.tsx
+
+In our <InlineCode>_app.tsx</InlineCode> file we import the CSS files that should be applues. We have 2, one that applies the values to root and is considered the default theme and the other that uses the new <InlineCode>tokens.data-theme.css</InlineCode> attribute to apply the theme based on the value set in the <InlineCode>data-theme</InlineCode> attribute.
+
+You can find the source code <Anchor showExternal={true} href="https://github.com/twilio-labs/paste/blob/main/packages/paste-website/src/pages/_app.tsx">here</Anchor>.
+
+<Box marginBottom="space60">
+  <CodeBlockWrapper>
+    <CodeBlockHeader>Importing CSS files</CodeBlockHeader>
+    <CodeBlock language='javascript' code={`import "@twilio-paste/design-tokens/dist/themes/twilio-dark/tokens.data-theme.css";
+import "@twilio-paste/design-tokens/dist/themes/twilio/tokens.custom-properties.css";
+`}/>
+  </CodeBlockWrapper>
+</Box>
+
+Note here that the import <InlineCode>"@twilio-paste/design-tokens/dist/themes/twilio/tokens.custom-properties.css"</InlineCode> is the default theme and the <InlineCode>"@twilio-paste/design-tokens/dist/themes/twilio-dark/tokens.data-theme.css"</InlineCode> is the theme that is applied when the <InlineCode>data-theme</InlineCode> attribute is set to <InlineCode>twilio-dark</InlineCode>.
+
+#### ThemeProvider
+
+In this same file we set the ThemeProvider to use CSS variables instead of the theme files we previously configured. Note that we only want to use CSS variables when encoutnering this issue and not as a standard. For all other use cases we recommend using the <InlineCode>theme</InlineCode> prop.
```

Comment:
```suggestion
In this same file we set the ThemeProvider to use CSS variables instead of the theme files we previously configured. Note that we only want to use CSS variables when encountering this issue and not as a standard. For all other use cases we recommend using the <InlineCode>theme</InlineCode> prop.
```